metadata_backend/api/handlers/folder.py

```diff
@@ -22,6 +23,135 @@
+            study = {
+                "attributes": {
+                    "doi": _study_doi,
+                    "prefix": _study_doi.split("/")[0],
+                    "suffix": _study_doi.split("/")[1],
+                    "types": {
+                        "bibtex": "misc",
+                        "citeproc": "collection",
+                        "schemaOrg": "Collection",
+                        "resourceTypeGeneral": "Collection",
+                    },
+                    "url": _study["url"],
+                    "identifiers": [_study["identifier"]],
+                },
+                "id": _study_doi,
+                "type": "dois",
+            }
+
+            study.update(_general_info)
+
+            _datasets = folder["extraInfo"]["datasetIdentifiers"]
+            for ds in _datasets:
+                _doi = ds["identifier"]["doi"]
+                _tmp = {
+                    "attributes": {
+                        "doi": _doi,
+                        "prefix": _doi.split("/")[0],
+                        "suffix": _doi.split("/")[1],
+                        "types": {
+                            "ris": "DATA",
+                            "bibtex": "misc",
+                            "citeproc": "dataset",
+                            "schemaOrg": "Dataset",
+                            "resourceTypeGeneral": "Dataset",
+                        },
+                        "url": ds["url"],
+                        "identifiers": [ds["identifier"]],
+                    },
+                    "id": _doi,
+                    "type": "dois",
+                }
+                _tmp.update(_general_info)
+
+                # A Dataset is described by a Study
+                if "relatedIdentifiers" not in _tmp["attributes"]:
+                    _tmp["attributes"]["relatedIdentifiers"] = []
+
+                _tmp["attributes"]["relatedIdentifiers"].append(
+                    {
+                        "relationType": "IsDescribedBy",
+                        "relatedIdentifier": _study_doi,
+                        "resourceTypeGeneral": "Collection",
+                        "relatedIdentifierType": "DOI",
+                    }
+                )
+
+                datasets.append(_tmp)
+
+                # A Study describes a Dataset
+                if "relatedIdentifiers" not in study["attributes"]:
+                    study["attributes"]["relatedIdentifiers"] = []
+
+                study["attributes"]["relatedIdentifiers"].append(
+                    {
+                        "relationType": "Describes",
+                        "relatedIdentifier": _doi,
+                        "resourceTypeGeneral": "Dataset",
+                        "relatedIdentifierType": "DOI",
+                    }
+                )
+        except Exception as e:
+            reason = f"Could not construct DOI data, reason: {e}"
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason)
```

Comment:
```suggestion
        except (KeyError, TypeError) as e:
            reason = f"Could not construct DOI data, reason: {e}"
            LOG.error(reason)
            raise web.HTTPBadRequest(reason=reason)
        except Exception as e:
            # unknown error
            reason = f"Could not construct DOI data, reason: {e}"
            LOG.error(reason)
            raise web.HTTPInternalServerError(reason=reason)
```
Consider catching exceptions explicitly and leaving general exception for unknown cases (not necessary, just something that came to mind)

Reply:
this will be handled in a follow-up PR, as more changes are needed